rust/datafusion/src/datasource/parquet.rs

```diff
@@ -41,10 +41,26 @@ impl ParquetTable {
     pub fn try_new(path: &str) -> Result<Self> {
         let parquet_exec = ParquetExec::try_new(path, None, 0)?;
         let schema = parquet_exec.schema();
+
+        let metadata = parquet_exec.metadata();
+        let num_rows = metadata
+            .row_groups()
+            .iter()
+            .map(|rg| rg.num_rows() as usize)
+            .sum();
+        let total_byte_size = metadata
+            .row_groups()
+            .iter()
+            .map(|rg| rg.total_byte_size() as usize)
```

Comment:
I think the as usize could move to below (`Some(num_rows as usize)`).

Than it could use `.map(RowGroupMetaData::total_byte_size)` as well?

Reply:
I am not familiar with this `.map(RowGroupMetaData::total_byte_size)` convention. do you know what this is called so I can look up the Rust docs?

Reply:
The way it is now is called "closure" and it could be replaced by a the name of the function instead. There is something on the clippy docs:
https://rust-lang.github.io/rust-clippy/master/#redundant_closure

Reply:
Thanks. I have updated the PR. I should have recognised this from Scala.